src/server/components/lists/controllers/index.ts

```diff
@@ -1,301 +1,15 @@
-import {
-  funeralDirectorsQuestionsSequence,
-  searchFuneralDirectors,
-} from "server/components/lists/searches/funeral-directors";
-import {
-  searchTranslatorsInterpreters,
-  translatorsInterpretersQuestionsSequence,
-} from "server/components/lists/searches/translators-interpreters";
-import type { LanguageRows } from "server/models/listItem/providers/types";
-import serviceName from "server/utils/service-name";
+import { getServiceTypeName } from "./../helpers";
 import { sendManualActionNotificationToPost } from "server/services/govuk-notify";
 
-export async function listsPostController(req: Request, res: Response, next: NextFunction): Promise<void> {
-  let params = getAllRequestParams(req);
-  // clean parameters
-  params = preProcessParams(params, req);
-  const { country, newLanguage } = params;
-  const { continueButton } = req.body;
-  let languagesProvided = params.languagesProvided ?? "";
-  let { serviceType } = params;
-  const serviceTypeName = getServiceTypeName(serviceType);
-
-  if (country && serviceType) {
-    try {
-      serviceType = serviceTypeName as ServiceType;
-      params = { ...params, serviceType };
-      const countryName: string = formatCountryParam(country);
-
-      const countryHasListItems = await some(countryName as CountryName, serviceType);
-      let redirectLink: string | undefined;
-
-      if (!countryHasListItems) {
-        switch (serviceType) {
-          case ServiceType.lawyers:
-            redirectLink = getCountryLawyerRedirectLink(countryName as CountryName);
-            break;
-          case ServiceType.covidTestProviders:
-            redirectLink = `${listsRoutes.privateBeta}?serviceType=${ServiceType.covidTestProviders}`;
-            break;
-          case ServiceType.funeralDirectors:
-            redirectLink = getCountryFuneralDirectorsRedirectLink(countryName as CountryName);
-            break;
-          case ServiceType.translatorsInterpreters:
-            redirectLink = getCountryTranslatorsInterpretersRedirectLink(countryName as CountryName);
-            break;
-          default:
-            redirectLink = undefined;
-        }
-
-        if (redirectLink !== undefined) {
-          res.redirect(redirectLink);
-          return;
-        }
-      }
-    } catch (error) {
-      next(error);
-      return;
-    }
-  }
-
-  if (newLanguage) {
-    languagesProvided = setLanguagesProvided(newLanguage, languagesProvided as string);
-    params.languagesProvided = languagesProvided;
-  }
-
-  if (params?.continueButton) {
-    delete params.continueButton;
-  }
-
-  const queryString = queryStringFromParams(params);
-  let url = `${listsRoutes.finder}?${queryString}`;
-  const languagesPopulated = !!continueButton;
-
-  if (languagesPopulated && params.languagesProvided) {
-    url = url.concat(`&languagesPopulated=true`);
-  }
-
-  res.redirect(url);
-}
-
-export function listsGetController(req: Request, res: Response): void {
```

Comment:
this controller did way too much. 

It's handling the languagesProvided param - even though it looked like questionaire.ts should've done that, it also looked like it was validating languages even though there was a fn for that?

same with servicesProvided.

imo most of the "lets" here didn't need to be declared as let. They seem fine as const?..